Sources/IndexStoreDB/IndexStoreDB.swift

```diff
@@ -80,6 +80,49 @@ public final class IndexStoreDB {
       return body(SymbolOccurrence(occur))
     }
   }
+
+  public func forEachCanonicalSymbolOccurrenceByName(symbolName: String, body: @escaping (SymbolOccurrence) -> Bool) {
+    indexstoredb_for_each_canonical_symbol_occurence_by_name(impl, symbolName) { (occur) -> Bool in
+      body(SymbolOccurrence(occur))
+    }
+  }
+
+  public func forEachCanonicalSymbolOccurrence(
+    containing pattern: String,
+    anchorStart: Bool,
+    anchorEnd: Bool,
+    subsequence: Bool,
+    ignoreCase: Bool,
+    body: @escaping (SymbolOccurrence) -> Bool
+  ){
+    indexstoredb_for_each_canonical_symbol_occurence_containing_pattern(
+      impl,
+      pattern,
+      anchorStart,
+      anchorEnd,
+      subsequence,
+      ignoreCase
+    ) { occur in
+      body(SymbolOccurrence(occur))
+    }
+  }
+
+  public func findSymbols(matching query: String) -> [SymbolOccurrence] {
```

Comment:
This method seems like it's fairly specific to what you want to do in sourcekit-lsp and the name doesn't make it clear what it's doing (e.g. skipping system occurrences?).  I suggest we remove it from the index API and just call `forEachCanonicalSymbolOccurrence(containing:...)` from `sourcekit-lsp`.